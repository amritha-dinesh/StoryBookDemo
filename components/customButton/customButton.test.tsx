import * as React from "react";
import { render } from "@testing-library/react-native";
import CustomButton from "./customButton";

it("renders text button by default", () => {
  const tree = render(<CustomButton buttonTitle="helloworld" />).toJSON();
  expect(tree).toMatchSnapshot();
});
it("renders text button with mode", () => {
  const tree = render(
    <CustomButton buttonTitle="helloworld" mode="text" />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it("renders contained button with mode", () => {
  const tree = render(
    <CustomButton mode="contained" buttonTitle="helloworld" />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it("renders disabled button", () => {
  const tree = render(
    <CustomButton disabled buttonTitle="helloworld" />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
it("renders button with custom testID", () => {
  const tree = render(
    <CustomButton testID={"custom:testID"} buttonTitle="helloworld" />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it("renders button with an accessibility label", () => {
  const tree = render(<CustomButton accessibilityLabel={"label"} />).toJSON();

  expect(tree).toMatchSnapshot();
});

it("renders button with an accessibility hint", () => {
  const tree = render(<CustomButton accessibilityHint={"hint"} />).toJSON();
  expect(tree).toMatchSnapshot();
});
it("renders button with an icon ", () => {
  const tree = render(<CustomButton icon={true} />).toJSON();
  expect(tree).toMatchSnapshot();
});
it("renders button with an indicator ", () => {
  const tree = render(<CustomButton loading={true} />).toJSON();
  expect(tree).toMatchSnapshot();
});
