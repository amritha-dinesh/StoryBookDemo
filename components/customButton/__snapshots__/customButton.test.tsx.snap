// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders button with an accessibility hint 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <View
    accessibilityHint="hint"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "transparent",
        "borderColor": "gray",
        "borderRadius": 0,
        "borderWidth": 0,
        "opacity": 1,
      }
    }
    testID="button"
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessibilityHint="hint-text"
        accessibilityLabel="undefined-text"
        style={
          [
            {
              "alignItems": "center",
              "justifyContent": "center",
              "padding": 10,
            },
            undefined,
            {
              "color": "#000000",
            },
          ]
        }
        testID="button-text"
      />
    </View>
  </View>
</View>
`;

exports[`renders button with an accessibility label 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <View
    accessibilityLabel="label"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "transparent",
        "borderColor": "gray",
        "borderRadius": 0,
        "borderWidth": 0,
        "opacity": 1,
      }
    }
    testID="button"
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessibilityHint="undefined-text"
        accessibilityLabel="label-text"
        style={
          [
            {
              "alignItems": "center",
              "justifyContent": "center",
              "padding": 10,
            },
            undefined,
            {
              "color": "#000000",
            },
          ]
        }
        testID="button-text"
      />
    </View>
  </View>
</View>
`;

exports[`renders button with an icon  1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "transparent",
        "borderColor": "gray",
        "borderRadius": 0,
        "borderWidth": 0,
        "opacity": 1,
      }
    }
    testID="button"
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <View
        testID="button-icon-container"
      >
        <Image
          source={
            {
              "testUri": "../../../src/assets/settings.png",
            }
          }
          style={
            {
              "height": 20,
              "marginLeft": 10,
              "width": 20,
            }
          }
        />
      </View>
      <Text
        accessibilityHint="undefined-text"
        accessibilityLabel="undefined-text"
        style={
          [
            {
              "alignItems": "center",
              "justifyContent": "center",
              "padding": 10,
            },
            undefined,
            {
              "color": "#000000",
            },
          ]
        }
        testID="button-text"
      />
    </View>
  </View>
</View>
`;

exports[`renders button with an indicator  1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "transparent",
        "borderColor": "gray",
        "borderRadius": 0,
        "borderWidth": 0,
        "opacity": 1,
      }
    }
    testID="button"
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <ActivityIndicator
        color="black"
        style={
          {
            "height": 20,
            "marginLeft": 10,
            "width": 20,
          }
        }
      />
      <Text
        accessibilityHint="undefined-text"
        accessibilityLabel="undefined-text"
        style={
          [
            {
              "alignItems": "center",
              "justifyContent": "center",
              "padding": 10,
            },
            undefined,
            {
              "color": "#000000",
            },
          ]
        }
        testID="button-text"
      />
    </View>
  </View>
</View>
`;

exports[`renders button with custom testID 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "transparent",
        "borderColor": "gray",
        "borderRadius": 0,
        "borderWidth": 0,
        "opacity": 1,
      }
    }
    testID="custom:testID"
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessibilityHint="undefined-text"
        accessibilityLabel="undefined-text"
        style={
          [
            {
              "alignItems": "center",
              "justifyContent": "center",
              "padding": 10,
            },
            undefined,
            {
              "color": "#000000",
            },
          ]
        }
        testID="custom:testID-text"
      >
        helloworld
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders contained button with mode 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": undefined,
        "borderColor": "gray",
        "borderRadius": 5,
        "borderWidth": 1,
        "opacity": 1,
      }
    }
    testID="button"
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessibilityHint="undefined-text"
        accessibilityLabel="undefined-text"
        style={
          [
            {
              "alignItems": "center",
              "justifyContent": "center",
              "padding": 10,
            },
            undefined,
            {
              "color": "#000000",
            },
          ]
        }
        testID="button-text"
      >
        helloworld
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders disabled button 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": true,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "transparent",
        "borderColor": "transparent",
        "borderRadius": 0,
        "borderWidth": 0,
        "opacity": 1,
      }
    }
    testID="button"
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessibilityHint="undefined-text"
        accessibilityLabel="undefined-text"
        style={
          [
            {
              "alignItems": "center",
              "justifyContent": "center",
              "padding": 10,
            },
            undefined,
            {
              "color": "#000000",
            },
          ]
        }
        testID="button-text"
      >
        helloworld
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders text button by default 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "transparent",
        "borderColor": "gray",
        "borderRadius": 0,
        "borderWidth": 0,
        "opacity": 1,
      }
    }
    testID="button"
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessibilityHint="undefined-text"
        accessibilityLabel="undefined-text"
        style={
          [
            {
              "alignItems": "center",
              "justifyContent": "center",
              "padding": 10,
            },
            undefined,
            {
              "color": "#000000",
            },
          ]
        }
        testID="button-text"
      >
        helloworld
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders text button with mode 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "transparent",
        "borderColor": "gray",
        "borderRadius": 0,
        "borderWidth": 0,
        "opacity": 1,
      }
    }
    testID="button"
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessibilityHint="undefined-text"
        accessibilityLabel="undefined-text"
        style={
          [
            {
              "alignItems": "center",
              "justifyContent": "center",
              "padding": 10,
            },
            undefined,
            {
              "color": "#000000",
            },
          ]
        }
        testID="button-text"
      >
        helloworld
      </Text>
    </View>
  </View>
</View>
`;
